core/model/src/main/java/org/eclipse/rdf4j/model/datatypes/XmlDatatypeEnum.java

```diff
@@ -0,0 +1,213 @@
+package org.eclipse.rdf4j.model.datatypes;
+
+import java.util.HashMap;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.xml.datatype.Duration;
+import javax.xml.datatype.XMLGregorianCalendar;
+
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.model.vocabulary.XSD;
+
+import com.google.common.collect.Sets;
+
+public enum XmlDatatypeEnum {
```

Comment:
I really dislike having the `Enum` suffix on the class name. How about we move this enum to an inner type of `XSD` and call it `Datatype` instead? You'd refer to it as `XSD.Datatype.DURATION` etc.

Reply:
I followed your recommendation here and it looks pretty decent. Thanks!